assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.js

```diff
@@ -306,14 +226,50 @@ const SearchFunnelWidget = ( { Widget, WidgetReportError } ) => {
 			);
 		}
 	);
-	const analyticsVisitorsOverviewAndStatsError = useSelect( ( select ) => {
-		if ( ! isAnalyticsConnected || showRecoverableAnalytics ) {
+
+	const analyticsLoading = useSelect( ( select ) => {
+		if (
+			! isAnalyticsConnected ||
+			! canViewSharedAnalytics ||
+			showRecoverableAnalytics
+		) {
 			return false;
 		}
 
-		return select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [
-			analyticsVisitorsOverviewAndStatsArgs,
-		] );
+		return (
+			! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [
+				analyticsOverviewArgs,
+			] ) ||
+			! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [
+				analyticsStatsArgs,
+			] ) ||
+			! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [
+				analyticsVisitorsOverviewAndStatsArgs,
+			] ) ||
+			! select( MODULES_ANALYTICS ).hasFinishedResolution(
+				'getGoals',
+				[]
+			)
+		);
+	} );
+
+	const analyticsError = useSelect( ( select ) => {
+		if ( ! isAnalyticsConnected || showRecoverableAnalytics ) {
+			return null;
+		}
+
+		return (
+			select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [
+				analyticsOverviewArgs,
+			] ) ||
+			select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [
+				analyticsStatsArgs,
+			] ) ||
+			select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [
+				analyticsVisitorsOverviewAndStatsArgs,
+			] ) ||
+			select( MODULES_ANALYTICS ).getErrorForSelector( 'getGoals', [] )
+		);
```

Comment:
The same point applies here:

```suggestion
		const { getErrorForSelector } = select( MODULES_ANALYTICS );

		return (
			getErrorForSelector( 'getReport', [ analyticsOverviewArgs ] ) ||
			getErrorForSelector( 'getReport', [ analyticsStatsArgs ] ) ||
			getErrorForSelector( 'getReport', [
				analyticsVisitorsOverviewAndStatsArgs,
			] ) ||
			getErrorForSelector( 'getGoals', [] )
		);
```

Reply:
Updated ✅